catalog/secrets/api/src/main/java/org/projectnessie/catalog/secrets/ExpiringTokenSecret.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.catalog.secrets;
+
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+
+/** Represents a "token" with an optional expiration. */
+public interface ExpiringTokenSecret extends Secret {
```

Comment:
Extend `TokenSecret`?

Reply:
It's gone now :)

Reply:
Nit: maybe rename this one to `TokenSecret`?